src/main/kotlin/org/rust/lang/refactoring/introduceVariable/RsNameSuggestions.kt

```diff
@@ -91,9 +104,9 @@ private fun nameForCall(expr: RsCallExpr): List<String> {
     return listOf(pathElement.text)
 }
 
-private fun findNamesInLocalScope(expr: PsiElement): List<String> {
+private fun findNamesInLocalScope(expr: PsiElement): HashSet<String> {
     val blockScope = expr.ancestorOrSelf<RsBlock>()
     val letDecls = blockScope?.descendantsOfType<RsLetDecl>().orEmpty()
 
-    return letDecls.mapNotNull { it.pat?.text }
+    return letDecls.asSequence().mapNotNull { it.pat?.text }.toHashSet()
```

Comment:
Hahahaha

Reply:
let (i, x) = (1, 2);

Reply:
I still think that in such cases we should use `mapNotNullTo`